institutions_ui/src/app/add-institution/page.tsx

```diff
@@ -112,8 +114,26 @@ export default function AddInstitution() {
         setLongitude('');
         setLatitude('');
         setDisabled(false);
+
+        if(shouldRedirect) {
+          router.push('/')
+        }
+        
       } else {
-        alert('Failed to add institution. Please try again.');
+        const error = await response.json();
+        let errorMessage = 'Error updating institution';
+        
+        // Handle array error responses (like 422 validation errors)
+        if (error.detail && Array.isArray(error.detail)) {
+          errorMessage = error.detail
+            .map((err: any) => err.msg || JSON.stringify(err))
+            .join('\n');
+        } else if (error.detail && typeof error.detail === 'string') {
+          // Handle string error messages
+          errorMessage = error.detail;
+        }
+        
+        alert(errorMessage);
```

Comment:
Pull this error handling code into a function and use it to handle errors on the update page as well. 

Also, note that response.json() could throw a error if the response isn't a json.